Adopt safer CPP in AccessibilityTableCell.cpp

jelee53 opened the pull request:
#### 18050cfe16ef556f32f16437b6cdd12285473c28
<pre>
Adopt safer CPP in AccessibilityTableCell.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=297629">https://bugs.webkit.org/show_bug.cgi?id=297629</a>
<a href="https://rdar.apple.com/158727073">rdar://158727073</a>

Reviewed by Tyler Wilcock.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines.">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines.</a>

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
(WebCore::AccessibilityNodeObject::checkedNode const):
* Source/WebCore/accessibility/AccessibilityTableCell.cpp:
(WebCore::AccessibilityTableCell::parentTable const):
(WebCore::AccessibilityTableCell::setRowIndex):
(WebCore::AccessibilityTableCell::setColumnIndex):
(WebCore::AccessibilityTableCell::titleUIElement const):
* Source/WebCore/rendering/RenderTableCell.h:
(WebCore::RenderTableCell::checkedTable const):

Canonical link: <a href="https://commits.webkit.org/299125@main">https://commits.webkit.org/299125@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fe5da065e4e10e89e7e7e4cf534ab7bc862b41a4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117949 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37627 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28262 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124096 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69984 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0356c464-0261-4b21-8d63-078027ea714e) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119827 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38320 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46209 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/89507 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/35c8112c-4067-45e1-b727-676996091503) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120901 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30516 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105766 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70001 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/691d5fe1-5b43-4473-bdf4-cd4bd4adc09d) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29583 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/23881 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67759 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99938 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24060 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127177 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44852 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33787 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/98174 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45213 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101990 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97962 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43357 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21345 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/41286 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18803 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44724 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50398 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44184 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47529 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45873 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/deacb414a75975f3918efbc774936c766689b78a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49630

jelee53 commented on Source/WebCore/accessibility/AccessibilityTable.cpp:
```diff
@@ -766,8 +769,8 @@ unsigned AccessibilityTable::computeCellSlots()
                 if (RefPtr axObject = cache->getOrCreate(node); axObject && needsToDescend(*axObject)) {
-                    for (const auto& child : axObject->childrenIncludingIgnored())
-                        processTableDescendant(child->node());
+                    for (const Ref<AXCoreObject>& child : axObject->childrenIncludingIgnored())
```
I'm not sure if this needs to be re-ref'd but we will see what the safer CPP checker says.

pvollan replied:
Is this change required by the Safer CPP bot?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/298da318b4dbec882cb2975a4fc701d369f7250f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49630

twilco commented:
The bug, Radar, and PR title should probably be updated if we're going to re-purpose it. It would be confusing to land as-is mentioning a class that doesn't exist (AccessibilityTable)

twilco commented on Source/WebCore/accessibility/AccessibilityTableCell.cpp:
```diff
@@ -341,31 +341,31 @@ AccessibilityObject* AccessibilityTableCell::titleUIElement() const
     // Table cells that are th cannot have title ui elements, since by definition
     // they are title ui elements
-    if (WebCore::elementName(node()) == ElementName::HTML_th)
+    if (WebCore::elementName(checkedNode().get()) == ElementName::HTML_th)
```
This pattern seems so strange...creating a CheckedPtr and immediately turning it into a raw pointer with `.get()` to pass into the function. Is this actually necessary? @rniwa do you know?

jelee53 replied:
I'm not Ryosuke but I have observed that this is an established pattern. 

This is an example from WebCore/rendering/RenderLayerCompositor.cpp:
```
void RenderLayerCompositor::paintContents(const GraphicsLayer* graphicsLayer, GraphicsContext& context, const FloatRect& clip, OptionSet<GraphicsLayerPaintBehavior>)
{
#if PLATFORM(MAC)
    LocalDefaultSystemAppearance localAppearance(m_renderView.useDarkAppearance());
#endif

    IntRect pixelSnappedRectForIntegralPositionedItems = snappedIntRect(LayoutRect(clip));
    if (graphicsLayer == layerForHorizontalScrollbar())
        paintScrollbar(RefPtr { m_renderView.frameView().horizontalScrollbar() }.get(), context, pixelSnappedRectForIntegralPositionedItems, m_viewBackgroundColor);
    else if (graphicsLayer == layerForVerticalScrollbar())
        paintScrollbar(RefPtr { m_renderView.frameView().verticalScrollbar() }.get(), context, pixelSnappedRectForIntegralPositionedItems, m_viewBackgroundColor);
```

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/aa048b46db6dcf5a314dd97a2f99b3afa2640d71. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49630

twilco commented on Source/WebCore/accessibility/AccessibilityObject.h:
```diff
@@ -973,6 +974,11 @@ inline AXObjectCache* AccessibilityObject::axObjectCache() const
     return m_axObjectCache.get();
 }
 
+inline CheckedPtr<AXObjectCache> AccessibilityObject::checkedAXObjectCache() const
```
Is this used? Can we delete it now?

twilco reviewed (approved):
r+ with `checkedAXObjectCache()` deleted

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4038b29f1c604306420c2fa2a0fcd55580546af3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49630

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4194382655d7dffe2d1179239a1ca06ff7c70011. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49630

jelee53 committed:
Adopt safer CPP in AccessibilityTableCell.cpp
https://bugs.webkit.org/show_bug.cgi?id=297629
rdar://158727073

Reviewed by Tyler Wilcock.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
(WebCore::AccessibilityNodeObject::checkedNode const):
* Source/WebCore/accessibility/AccessibilityTableCell.cpp:
(WebCore::AccessibilityTableCell::parentTable const):
(WebCore::AccessibilityTableCell::setRowIndex):
(WebCore::AccessibilityTableCell::setColumnIndex):
(WebCore::AccessibilityTableCell::titleUIElement const):
* Source/WebCore/rendering/RenderTableCell.h:
(WebCore::RenderTableCell::checkedTable const):

Canonical link: https://commits.webkit.org/299125@main
